packages/shared-components/src/components/core/onto-dropdown/onto-dropdown.tsx

```diff
@@ -0,0 +1,123 @@
+     */
+    @Prop() nameLabel: string;
+
+    /**
+     * Key for the dropdown button label, used with the translation service.
+     */
+    @Prop() nameLabelKey: string;
+
+    /**
+     * The tooltip text displayed for the dropdown button.
+     * Overrides `tooltipLabelKey` if provided.
+     */
+    @Prop() tooltipLabel: string;
+
+    /**
+     * Key for the tooltip label, used with the translation service.
+     */
+    @Prop() tooltipLabelKey: string;
+
+    /**
+     * Icon class for the main dropdown button.
+     */
+    @Prop() iconClass = '';
+
+    /**
+     * Array of dropdown options. Each option includes a value, label, tooltip, and optional icon class.
+     */
+    @Prop() items: DropdownOption[];
+
+    /**
+     * Event emitted when a dropdown item is selected.
+     * The event payload contains the value of the selected item.
+     */
+    @Event() valueChanged: EventEmitter<string>;
+
+    /**
+     * Listens for the Escape key keydown event globally.
+     * Closes the dropdown menu if the Escape key is pressed.
+     *
+     * @param ev The keyboard event triggered by the Escape key.
+     */
+    @Listen('keydown', {target: 'window'})
+    keydownListener(ev: KeyboardEvent): void {
+        if (ev.key === 'Escape') {
+            this.closeMenu();
+        }
+    }
+
+    /**
+     * Listens for mouse click events globally.
+     * Closes the dropdown menu if the click occurs outside the dropdown.
+     *
+     * @param ev The mouse event triggered by a click.
+     */
+    @Listen('click', {target: 'window'})
+    mouseClickListener(ev: PointerEvent): void {
+        const target: HTMLElement = ev.target as HTMLElement;
+        if (!target.closest('.onto-dropdown')) {
+            this.closeMenu();
+        }
+    }
+
+    render() {
+        return (
+            <div class={`onto-dropdown btn-group ${this.open ? 'open' : 'closed'}`}
+                 title={this.tooltipLabel ? this.tooltipLabel : TranslationService.translate(this.tooltipLabelKey)}>
+                <button class={"btn btn-secondary dropdown-toggle"}
+                        onClick={() => this.toggleComponent()}>
+                    <span class={this.iconClass}></span>
+                    <span
+                        class='button-name'>{this.nameLabel ? this.nameLabel : TranslationService.translate(this.nameLabelKey)}</span>
+                </button>
+
+                <div class="dropdown-menu pre-scrollable dropdown-menu-right">
+                    {this.items && this.items.map(item =>
+                        <a href="#" class='dropdown-item' title={TranslationService.translate(item.tooltip)}
+                           onClick={() => this.onSelect(item.value)}>
+                            <span class={'dropdown-option-icon ' + item.iconClass}></span>
+                            {item.label ?
+                                <span innerHTML={item.label}></span> : TranslationService.translate(item.labelKey)}
```

Comment:
I'd prefer to have consistent html structure in both cases here. Add a span tag around the translated label as well.

Reply:
Also, it'd be possible in practice to have html tags in the translated labels as well. So, in both cases the label should be injected via the innerHTML

Reply:
fixed